atom/browser/mac/atom_application_delegate.mm

```diff
@@ -87,15 +87,11 @@ - (BOOL)application:(NSApplication*)sender
   return atom::Browser::Get()->OpenFile(filename_str) ? YES : NO;
 }
 
-- (NSApplicationTerminateReply)applicationShouldTerminate:(NSApplication*)sender {
-  atom::Browser* browser = atom::Browser::Get();
-  if (browser->is_quiting()) {
-    return NSTerminateNow;
-  } else {
-    // System started termination.
-    atom::Browser::Get()->Quit();
-    return NSTerminateCancel;
-  }
+// We simply try to close the browser, which in turn will try to close the windows.
+// Termination can proceed if all windows are closed or window close can be cancelled
+// which will abort termination.
+- (void)tryToTerminateApp:(NSApplication*)app {
+  atom::Browser::Get()->Quit();
```

Comment:
It's possible this request could be received while the app is already quitting, we should probably still check that `browser->is_quitting()` is `false`

Reply:
`Browser::Quit` already does the check itself.